python/observation-publisher/publisher/prediction_log_consumer.py

```diff
@@ -34,22 +39,68 @@ def close(self):
         raise NotImplementedError
 
     def start_polling(
-        self, observation_sink: ObservationSink, inference_schema: InferenceSchema
+        self,
+        observation_sinks: List[ObservationSink],
+        inference_schema: InferenceSchema,
     ):
         try:
+            buffered_logs = []
+            buffered_max_duration_seconds = 60
+            buffer_start_time = datetime.now()
             while True:
                 logs = self.poll_new_logs()
                 if len(logs) == 0:
                     continue
-                df = log_batch_to_dataframe(logs, inference_schema)
-                observation_sink.write(df)
+                buffered_logs.extend(logs)
+                buffered_duration = (datetime.now() - buffer_start_time).seconds
+                if (
+                    len(buffered_logs) < self.buffer_capacity
+                    and buffered_duration < buffered_max_duration_seconds
+                ):
+                    continue
+                df = log_batch_to_dataframe(buffered_logs, inference_schema)
+                most_recent_prediction_timestamp = df[
+                    PREDICTION_LOG_TIMESTAMP_COLUMN
+                ].max()
+                MetricWriter().update_last_processed_timestamp(
+                    most_recent_prediction_timestamp
+                )
+                write_tasks = [
+                    Thread(target=sink.write, args=(df,)) for sink in observation_sinks
+                ]
+                for task in write_tasks:
+                    task.start()
+                for task in write_tasks:
+                    task.join()
                 self.commit()
+                buffered_logs = []
+                buffer_start_time = datetime.now()
         finally:
             self.close()
 
 
+@dataclass_json
+@dataclass
+class KafkaConsumerConfig:
+    topic: str
+    bootstrap_servers: str
+    group_id: str
+    batch_size: int = 100
+    poll_timeout_seconds: float = 1.0
+    additional_consumer_config: Optional[dict] = None
+
+
 class KafkaPredictionLogConsumer(PredictionLogConsumer):
-    def __init__(self, config: KafkaConsumerConfig):
+    def __init__(
+        self,
+        buffer_capacity: int,
```

Comment:
seems the buffer capacity is number of records that is being hold, why not size of the data that it holds?